### PR TITLE
PLAT-10892: return s3 file location or local server file path for path only requests coming from s3 supported packager

### PR DESCRIPTION
**Pull request by inbal-ben-david**
This change is [Reviewable](https://reviewable.io/reviews/kaltura/server/9419)

### CODE REVIEW

**Comment by inbal-ben-david** [comment]
@MosheMaorKaltura @yossipapi

File: alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php [hunk 1]
```diff
@@ -1778,5 +1780,100 @@ public static function getFileSyncFromPeriodicStorage($asset, $syncKey)
...
+		if(in_array($fileSync->getDc(), $localDcIds))
+		{
+			// use serve file local url
+			$dcConfig = kDataCenterMgr::getDcById($fileSync->getDc());
+			$path = '/' . strtolower($dcConfig['name']) . kDataCenterMgr::getInternalRemoteUrl($fileSync, false);
+		}
+		else
+		{
+			// use cloud file location
+			$path = self::CLOUD_S3_PREFIX . $fileSync->getFilePath();
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php:1847]: Try to get the prefix from the remote storage

File: alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php [hunk 2]
```diff
@@ -1778,5 +1780,100 @@ public static function getFileSyncFromPeriodicStorage($asset, $syncKey)
...
+	}
+
+	/**
+	 * get the object path in remote location or the serveFile path for local location
+	 *
+	 * @param $fileSync
+	 * @return string
+	 * @throws Exception
+	 */
+	public static function getPathByFileSync($fileSync)
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php:1835]: If the preferred storage ID is not local - user URL else user file.

File: alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php [hunk 3]
```diff
@@ -486,23 +482,23 @@ public function execute()
...
 			$path = '';
 			$parent_file_sync = null;
-			list ( $file_sync , $local )= kFileSyncUtils::getReadyFileSyncForKey( $syncKey , true, false );
-			if ( $file_sync )
+
+			if($preferredStorageId)
 			{
-				$allowRemotePath = false;
-				if(!$local)
+				$file_sync = kFileSyncUtils::getFileSyncByPreferredOrder($syncKey, $flavorAsset, $preferredStorageId);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php:488]: Rename - no Order

**Pull request merged**